sympy/combinatorics/permutations.py

```diff
@@ -443,7 +444,7 @@ def __init__(self, *args):
                 for k, v in args[0].items():
                     self[k] = v
                 return
-        args = [int(a) for a in args]
+        args = [int(a) for a in args if a >= 0]
```

Comment:
Having the -ve integers as elements leads to unusual behavior:
```
>>> c = Cycle(-4, 2)
>>> c
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/gxyd/Public/sympy/sympy/combinatorics/permutations.py", line 394, in __repr__
    cycles = Permutation(self).cyclic_form
  File "/home/gxyd/Public/sympy/sympy/combinatorics/permutations.py", line 1075, in cyclic_form
    while unchecked[array_form[j]]:
IndexError: list index out of range
>>> c.list()
[0, 1, -4]
```

I will look into what needs to be done for `Permutation` as well.